### PR TITLE
Upgrade to Gradle 5.0

### PR DESCRIPTION
**Pull request by SamCarlberg**
Update shadow plugin 2.0.1 -> 4.0.3
Remove source and javadoc jars from app artifact publishing

### CODE REVIEW

File: plugins/plugins.gradle.kts [hunk 1]
```diff
@@ -10,17 +10,18 @@ subprojects {
         }
         val sourceJar = task<Jar>("sourceJar") {
             description = "Creates a JAR that contains the source code."
-            from(java.sourceSets["main"].allSource)
+            from(project.sourceSets["main"].allSource)
             classifier = "sources"
         }
+        val javaPluginConvention = convention.plugins["java"] as JavaPluginConvention
```
Comment (SamCarlberg) [inline plugins/plugins.gradle.kts]: @JLLeitschuh This is ugly as hell, but I couldn't find a better way to get the output docs directory without hardcoding it

File: plugins/plugins.gradle.kts [hunk 2]
```diff
@@ -10,17 +10,18 @@ subprojects {
...
         val javadocJar = task<Jar>("javadocJar") {
             dependsOn("javadoc")
             description = "Creates a JAR that contains the javadocs."
-            from(java.docsDir)
+            from(javaPluginConvention.docsDir)
```
Comment (JLLeitschuh) [inline plugins/plugins.gradle.kts]: Out of curiosity, can you just use `from(tasks.named("javadoc")` here?

File: plugins/plugins.gradle.kts [hunk 3]
```diff
@@ -10,17 +10,18 @@ subprojects {
         }
         val sourceJar = task<Jar>("sourceJar") {
             description = "Creates a JAR that contains the source code."
-            from(java.sourceSets["main"].allSource)
+            from(project.sourceSets["main"].allSource)
             classifier = "sources"
         }
+        val javaPluginConvention = convention.plugins["java"] as JavaPluginConvention
         val javadocJar = task<Jar>("javadocJar") {
```
Comment (JLLeitschuh) [inline plugins/plugins.gradle.kts]: Use `tasks.registering` now that that task providers should work as an argument to `artifact`.

Comment (SamCarlberg) [reply plugins/plugins.gradle.kts]: Nope.

---
Using `tasks.register`:

```
Cannot convert the provided notation to an object of type MavenArtifact: task ':plugins:base:javadocJar'.
The following types/formats are supported:
  - Instances of MavenArtifact.
  - Instances of AbstractArchiveTask, for example jar.
  - Instances of PublishArtifact
  - Maps containing a 'source' entry, for example [source: '/path/to/file', extension: 'zip'].
  - Anything that can be converted to a file, as per Project.file()
```

---

Using `tasks.registering`:
```
Cannot convert the provided notation to an object of type MavenArtifact: org.gradle.kotlin.dsl.RegisteringDomainObjectDelegateProviderWithTypeAndAction@7600c166.
The following types/formats are supported:
  - Instances of MavenArtifact.
  - Instances of AbstractArchiveTask, for example jar.
  - Instances of PublishArtifact
  - Maps containing a 'source' entry, for example [source: '/path/to/file', extension: 'zip'].
  - Anything that can be converted to a file, as per Project.file()
```

File: plugins/plugins.gradle.kts (hunk 2 above)
Comment (SamCarlberg) [reply plugins/plugins.gradle.kts]: Yep!

File: plugins/plugins.gradle.kts (hunk 3 above)
Comment (JLLeitschuh) [reply plugins/plugins.gradle.kts]: Ah, I thought it had been fixed for 5.1. Aparently not.

File: plugins/plugins.gradle.kts [hunk 4]
```diff
@@ -10,17 +10,17 @@ subprojects {
...
             description = "Creates a JAR that contains the source code."
-            from(java.sourceSets["main"].allSource)
+            from(project.sourceSets["main"].allSource)
             classifier = "sources"
         }
         val javadocJar = task<Jar>("javadocJar") {
             dependsOn("javadoc")
             description = "Creates a JAR that contains the javadocs."
-            from(java.docsDir)
+            from(tasks.named("javadoc"))
```
Comment (JLLeitschuh) [inline plugins/plugins.gradle.kts]: 🎉

**Pull request merged**